File: Data Structures/Topic Wise/Tree/binary_search.cpp
#include<bits/stdc++.h>
using namespace std;

void binary()
{
    int arr[5]={2,4,6,7,9};
    int l=0, h=4, mid , key , ind=-1;
    cout<<"enter the to be search :: ";
    cin>>key;

    while (l<=h)
    {
        mid=(l+h)/2;
        if(key==arr[mid])
        {
            ind=mid;
            break;
        }
        else if(key>arr[mid])
        {
             l=mid+1;
        }
        else if(key<arr[mid])
        {
             h=mid-1;
        }
    }
    if(ind==-1)
    {
        cout<<"element Not found ";
    }
    else
    {
        cout<<"element is found at index :: "<<ind;
    }
}
int main()
{
    binary();
   
   return 0;
}

File: java/ApnaCollege/array/binarySearch.java
class binarySearch
{
     public static int binarySearch(int arr[],int key)
     {
          int l=0,h=arr.length-1;
          
          while(l<=h)
          {
              int mid=l+(h-l)/2;
              if(arr[mid]==key)
              {
                  return mid;
              }
              else if(arr[mid]>key)
              {
                  h=mid-1;
              }
              else
              {
                  l=mid+1;
              } 
          }
          return -1;
          
     }
     public static void main(String arg[])
     {
          
         int arr[]={1,3,5,6,7,9,10};
         int index=binarySearch(arr,11);
         System.out.print("index ::"+index);
     }
}
